config.yaml

```diff
@@ -6,11 +6,46 @@ options:
     description: The number of webserver worker processes for handling requests.
   webserver_threads:
     type: int
-    description: >
-      The number of worker threads for handling requests. Run each webserver worker with the specified number of threads.
+    description: Run each webserver worker with the specified number of threads.
   webserver_keepalive:
     type: int
-    description: The number of seconds for webserver to wait for requests on a Keep-Alive connection.
+    description: Time in seconds for webserver to wait for requests on a Keep-Alive connection.
   webserver_timeout:
     type: int
-    description: Webserver workers silent for more than this many seconds are killed and restarted.
+    description: Time in seconds to kill and restart silent webserver workers.
+  flask_env:
+    type: string
+    description: What environment the Flask app is running in, by default it's 'production'.
+  flask_debug:
+    type: boolean
+    description: Whether Flask debug mode is enabled.
+  flask_secret_key:
+    type: string
+    description: >
+      A secret key used for securely signing the session cookie and  for any other security related needs by your 
+      Flask application. This configuration will set the FLASK_SECRET_KEY environment variable.
+      Run `app.config.from_prefixed_env()` in your Flask application in order to receive this configuration.
+  flask_permanent_session_lifetime:
+    type: int
+    description: >
+      Time in seconds for the cookie to expire in the Flask application permanent sessions. This configuration 
+      will set the FLASK_PERMANENT_SESSION_LIFETIME environment variable. Run `app.config.from_prefixed_env()` 
+      in your Flask application in order to receive this configuration.
+  flask_application_root:
+    type: string
+    description: >
+      Path in which the application / web server is mounted. This configuration will set the FLASK_APPLICATION_ROOT 
+      environment variable. Run `app.config.from_prefixed_env()` in your Flask application in order to receive 
+      this configuration.
+  flask_session_cookie_secure:
+    type: boolean
+    description: >
+      Set the secure attribute in the Flask application cookies. This configuration will set the 
+      FLASK_SESSION_COOKIE_SECURE environment variable. Run `app.config.from_prefixed_env()` in your Flask application
+      in order to receive this configuration.
+  flask_preferred_url_scheme:
+    type: string
+    description: >
+      Scheme for generating external URLs when not in a request context in the Flask application. By default,
+      it's "HTTP" This configuration will set the FLASK_PREFERRED_URL_SCHEME environment variable. 
```

Comment:
Should this default to HTTPS?

Reply:
The default value is decided by Flask.

Reply:
Should we be more opinionated and make the default HTTPS?

Reply:
I think it's best not to altering flask configuration defaults to prevent user confusion.

Reply:
@gregory-schiano? Any thoughts? What are the impacts of changing to HTTPS, does that mean the dev server doesn't work? One of the problems with security is unsafe defaults, and this is potentially one

Reply:
I can change the default of FLASK_PREFERRED_URL_SCHEME to be HTTPS very easily. However, for flask_secret_key it requires a lot of efforts, since the secret key requires a persistent storage (peer relation) and we need to provide an action to reset secret key if there's a data breach. Considering the effort involved, we should evaluate whether the benefits outweigh the costs

Also, If we decide to implement charm generated secret, that will be out of the scope of this PR, we need to create another story for that.

Reply:
The default value of preferred_url_scheme has been changed to HTTPS, thanks!